Userland/Libraries/LibPDF/Document.cpp

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021, Matthew Olsson <mattco@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibPDF/Document.h>
+#include <LibPDF/Parser.h>
+
+namespace PDF {
+
+Document::Document(const ReadonlyBytes& bytes)
+    : m_parser(Parser({}, bytes))
+{
+    VERIFY(m_parser.perform_validation());
+    auto [xref_table, trailer] = m_parser.parse_last_xref_table_and_trailer();
```

Comment:
Is the `Parser` actually used after the constructor?
If not, could we rearrange things so that `Document` doesn't have to lug it around forever?

Reply:
Maybe not in this PR, but definitely in the future. The document will do a lot of lazy parsing, such as parsing pages only when the PDFViewer requests them.

Reply:
Gotcha, okay :)